lambda/events_id/get/index.js

```diff
@@ -2,104 +2,127 @@
     
     // Console log to CloudWatch for the event id provided in the path parameters.
     console.log("Path parameter, event_id: ", eventId);
-   
-    // parameters for DynamoDB request. 
-    const params = {
-        TableName: 'sprint2_testTable',
-        KeyConditionExpression: "#event_id = :event_id",
-        // Specify which attribute to search for
-        ExpressionAttributeNames:{
-            "#event_id": "event_id"
-        },
-        // Specify which attribute value {event id} to search for
-        ExpressionAttributeValues: {
-            ":event_id": eventId
-        },
-        // ProjectionExpression is for the request attributes of events in table 
-        ProjectionExpression: "event_id, user_name, activity_type, description, organization_name, location_name, location_address, contact_name, contact_phone, contact_email, start_date_time, end_date_time, frequency, cost, picture_url, min_age, max_age, disability_types, show_status, approver"
-    }//end of params
 
+    // this stores the http method used by the caller
+    //const httpMethod = event.httpMethod;
+    const httpMethod = event.context['http-method'];
     
-    try {
-        // Creating variable {data} to store the response from DynamoDB using the params created above
-        // documentClient.query() will search the entire table based on the parameters given.
-        const data = await documentClient.query(params).promise();
-        responseBody = data;
-
-        // Checking if the query returned with an event or found no matches
-        if(responseBody.Count == 0) {
-            //setting responseBody and statusCode on no results returned from DynamoDB
-            responseBody = 'Event not found';
-            statusCode = 404;
-            
-            // Log for no event found
-            console.log("Query did not find a matching event with the event_id provided. Status code: ", statusCode);
-        } else {
-            //setting responseBody and statusCode on successful return of event from DynamoDB
-            responseBody = JSON.stringify(responseBody.Items);
-            statusCode = 200;
-            
-            // Log for event found
-            console.log("Event found, status code: ", statusCode);
-        }
-        
-    } catch(err) {
-        responseBody = 'Failure to connect or retrieve from the database';
-        statusCode = 500;
-        
-        // Log for error with DynamoDB query
-        console.error("Unsuccessful connection to DynamoDB. Status code: " + statusCode);
-        console.error("Error: " + err);
-    }//end of catch
+    //Log calling API type and endpoint
+    console.log("API-Endpoint called: " + httpMethod + " " + event.context['resource-path']);
+    
+    //Log calling IP
+    console.log("Calling-IP: " + event.params.header['X-Forwarded-For']);
 
     // Create response to return with statusCode, headers, and responseBody. 
     const response = {
```

Comment:
I'm pretty sure this code isn't doing anything, apart from being logged to Cloudwatch. Headers are being set by API Gateway, so it's probably save to remove lines 52-57 and update Cloudwatch logs accordingly

Reply:
Removed the code about header in the response. Keeping response as it is being used throughout the code and for my logging.